eclipse-theia/theia
# Fix debug launch failures from variable resolution

#### What it does

Fix debug launch failures from variable resolution

The debug session manager relies on the variable resolving services
to replace variables present in the launch configuration.

Some of these variables are resolved by interactive commands towards
the user, however if the user press escape or ignores a prompt, the
variable resolution will not take place and if the flow continues,
errors will be displayed to the user (unnecessarily).

The solution before this fix was relying on a check on the successful
resolution of all the variables i.e. non are 'undefined', however
there are cases were this happens and the flow shall continue.

This fix, replaces the check for the presence of `escape` entries in
an array, this array is populated with the command ids that were not
successfully resolved i.e. giving a result of 'null'.

Signed-off-by: Alvaro Sanchez-Leon <alvaro.sanchez-leon@ericsson.com>



#### How to test


1) Download and build the following test project 
https://github.com/alvsan09/hello_express

2) Open the project in theia

3) From the debug view launch the configuration `Attach to node process`, 
- Verify that a list of process is present
- Click outside the list area or the escape key
- Make sure that not Error dialogs pop up

4) Launch the configuration `Launch Backend`
5) Launch the configuration `Launch Frontend`
- Verify that a chrome browser opens successfully with the `Hello World` message.

![launch_with_undefined_variables](https://user-images.githubusercontent.com/76971376/175425026-3f0e83a8-ca4f-44b1-a198-1ec9aa040e2e.gif)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
